hub/hub.go

```diff
@@ -111,6 +123,7 @@ func (c *EventHubClient) Write(ctx context.Context, samples model.Samples) error
 
 		if err := c.hub.SendBatch(ctx, eventhub.NewEventBatchIterator(events...)); err != nil {
 			log.ErrorObj(err).Msg("send event batch")
+			return err
```

Comment:
@bryanklewis I had to return the err here to force the connection to be closed. Otherwise, the software was still using the connection even though it got a TCP RST and the socket was closed. I reproduced this issues using a linux box and iptables to give a TCP reset to the application on demand. Like:

`iptables -A OUTPUT -p tcp -d 52.168.117.xx/32 -j REJECT --reject-with tcp-reset`

Reply:
This makes sense and is an awesome catch. Most other Azure services, like app hosting plans or databases, when you change service class it often re-IPs. It makes sense that with auto-inflate enable we would see the same behavior happen on the Azure side. Thank-you so much for taking the time to track it down and provide a fix!

Reply:
thank you!